### PR TITLE
Notify if dependency releases a major version

### PR DESCRIPTION
**Pull request by kshitij-microsoft**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by PratibhaShrivastav18** [comment]
@kshitij-microsoft , can you resolve the copilot reviews? They seem helpful

File: .github/workflows/dependecy-checker.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Check the dependencies for major version updates in azure-ai-ml
```
Comment (weshaggard) [inline .github/workflows/dependecy-checker.yml]: What is the context on this github action? Who do you expect to be watching the results? Should this be in one of the existing pipelines instead?

Comment (kshitij-microsoft) [reply .github/workflows/dependecy-checker.yml]: We are trying to get a notification in case there are any update in the major version of the dependencies being used by azure-ai-ml package. azure-ai-ml team will be taking care of the notifications.

File: sdk/ml/azure-ai-ml/scripts/major_updates.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import ast
+import sys
+
```
Comment (jayesh-tanna) [inline sdk/ml/azure-ai-ml/scripts/major_updates.py:3]: Can you please add unit test cases for all cases and edge cases?

File: sdk/ml/azure-ai-ml/scripts/major_updates.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        all_version = list_versions(dep.name)
+        if len(all_version) > 1 and is_major_update(all_version[1], all_version[0]):
+            major_updates.append(
+                f"Current config of `{dep}` and major version updated from {dep.name}: {all_version[1]} → {all_version[0]}"
+            )
+    if major_updates:
+        # body = "\n".join(major_updates) # Uncomment this line if you want to send an email or notification
+        print("Major Dependency Updates Found:")
+        print(major_updates)
+        sys.exit(1)
```
Comment (jayesh-tanna) [inline sdk/ml/azure-ai-ml/scripts/major_updates.py:66]: As per our discussion, we want to create GH issue/ICM etc. is it included in this implementation?

Comment (kshitij-microsoft) [reply sdk/ml/azure-ai-ml/scripts/major_updates.py:66]: no, that will be taken care in the later iterations.

File: sdk/ml/azure-ai-ml/scripts/major_updates.py (hunk 2 above)
Comment (kshitij-microsoft) [reply sdk/ml/azure-ai-ml/scripts/major_updates.py:3]: This script is bound to change with the new suggestions by the team. Once those changes are made, will add the tests cases as well.

**Review by jayesh-tanna — APPROVED** [review]

**Comment by kshitij-microsoft** [comment]
@Azure/azure-sdk-eng Need your approval here.

**Review by achauhan-scc — APPROVED** [review]
